src/quantem/diffractive_imaging/ptychography_base.py

```
@@ -476,12 +520,15 @@ def probe_model(self, model: ProbeModelType | type):
         self._probe_model = cast(
             ProbeModelType, model
         )  # have before so that energy available to set initial probe
-        self._probe_model.set_initial_probe(
-            self.roi_shape,
-            self.reciprocal_sampling,
-            self.dset.mean_diffraction_intensity,
-            device=self.device,
-        )
+        try:
+            self._probe_model.set_initial_probe(
+                self.roi_shape,
+                self.reciprocal_sampling,
+                self.dset.mean_diffraction_intensity,
+                device=self.device,
+            )
+        except Exception:
+            pass
```

why is this allowed to fail silently again?

---

This only fails if the dset hasn't been preprocessed, and is then called in ptycho.preprocess instead. I'm updating it to not be an explicit if/else call rather than try/except with a comment to explain.